chainstate/types/src/block_index.rs

```diff
@@ -36,7 +36,12 @@ pub struct BlockIndex {
     #[codec(compact)]
     /// The total number of transactions up to this block
     chain_transaction_count: u128,
+    /// The block status.
     status: BlockStatus,
+    /// If true, this block index will stay in the db forever; otherwise, it may be deleted
```

Comment:
It's not forever. It's until the reorg depth is reached (when pruning is implemented); then, if it's mainchain, it'll stay, otherwise it'll be deleted.